lib/src/login/login.dart

```diff
@@ -191,13 +196,44 @@ class _LoginState extends State<Login> {
     List<TextSpan> textParts = [];
     textParts.add(TextSpan(text: formattedAgreeString.substring(spanBoundary, termsConditionsStartIndex)));
     spanBoundary = termsConditionsStartIndex;
-    textParts.add(UrlTextSpan(url: null, text: termsAndConditions));
+    textParts.add(TextSpan(
+      text: termsAndConditions,
+      style: TextStyle(color: accent, fontWeight: FontWeight.bold),
+      recognizer: TapGestureRecognizer()
+        ..onTap = () {
+          _navigation.push(
+              context,
+              MaterialPageRoute(
+                builder: (context) => Privacy(
+                  htmlText: _terms,
+                ),
+              ));
+        },
+    ));
     spanBoundary = termsConditionsEndIndex;
     textParts.add(TextSpan(text: formattedAgreeString.substring(spanBoundary, privacyPolicyStartIndex)));
     spanBoundary = privacyPolicyStartIndex;
-    textParts.add(UrlTextSpan(url: null, text: privacyPolicy));
+    textParts.add(TextSpan(
+      text: privacyPolicy,
+      style: TextStyle(color: accent, fontWeight: FontWeight.bold),
+      recognizer: TapGestureRecognizer()
+        ..onTap = () {
+          _navigation.push(
+              context,
+              MaterialPageRoute(
+                builder: (context) => Privacy(
+                  htmlText: _privacy,
+                ),
+              ));
+        },
+    ));
     spanBoundary = privacyPolicyEndIndex;
     textParts.add(TextSpan(text: formattedAgreeString.substring(spanBoundary)));
     return textParts;
   }
+
+  _loadHtmlFiles() async {
+    _terms = await rootBundle.loadString("assets/html/terms.html");
+    _privacy = await rootBundle.loadString("assets/html/privacypolicy.html");
+  }
```

Comment:
Please move the loading logic to privacy.dart. We should just provide the asset path and privacy.dart should load / parse the content. I would also suggest renaming the privacy.dart to web_asset.dart or something similar, so we can reuse it for other HTML content later, without getting confused. ;)